### ISSUE TITLE
opam list --depends-on sometimes doesn't care about the version number

### ISSUE THREAD

**Issue by kit-ty-kate** [ocaml/opam#3250]
This bug is probably related to https://github.com/ocaml/opam/issues/3229.

Sometimes when requesting the list of packages that depends on a given version of a package, opam list is going to give wrong answers.
For example, here if we requests all the packages that depends on ```cstruct.3.2.0```, it will give ```asn1-combinators.0.1.0``` which specifically requests ```"cstruct" {>= "1.2.0" & < "1.6.0"}```:
```
opam list --all-version --installable --available --depends-on cstruct.3.2.0 | grep asn1-combinators.0.1.0
```
But it's not always the case. For instance here if we ask for the list of packages that depends on ```angstrom.0.8.0```, ```type-beat.0.1``` will not be given even though it depends on ```"angstrom" {< "0.7.0"}``` (however ```type-beat``` /= ```typebeat```…):
```
opam list --all-version --installable --depends-on angstrom.0.8.0 | grep type-beat.0.1
```

PS: ```--installable``` and ```--available``` doesn't seems to have much effect regarding this issue

**Comment by trefis** [comment]
Gentle ping on this issue.
I believe the CI for opam-repository is doing frequently doing work that it shouldn't because of this. It's not very environmentally friendly.

**Comment by AltGr** [comment]
This should be fixed by 24b5874, which is in a PR but unmerged yet
